### PR TITLE
Improve Python api docstring rtype resolution

### PR DESCRIPTION
**Pull request by mb-a**
This is a small helpful change to aid in auto-resolving types in PyCharm (and I imagine other parsers will also enjoy this).

My initial attempt was to add `lusid.models.` to everything by checking `returnTypeIsPrimitive`:
`{{^returnTypeIsPrimitive}}{{packageName}}.models.{{/returnTypeIsPrimitive}}`
This was found by debugging with `openapi-generator-cli.jar [...] --global-property debugOperations=true`.

Unfortunately that wouldn't cover something like `lusid.api.legal_entities_api.get_all_legal_entity_access_metadata`, which has a returnType `dict(str, list[AccessMetadataValue])`, so the alternative is to copy the csharp generator behaviour and explicitly import all the imports next to `hasImport` at the top of the api file.
It's a little unsightly, but it vastly improves my development experience with the Lusid SDK.

Also changes the `_with_http_info` template to omit `tuple` from the rtype, which also helps with auto-resolution.

As a side note, `dict(str, list[AccessMetadataValue])` doesn't auto-resolve in PyCharm, but `dict[str, list[AccessMetadataValue]]` does -- however I suspect that changing this will break a few openapi-generator test cases.

### CODE REVIEW

File: python/generate/templates/api.mustache [hunk 1]
```diff
@@ -13,7 +13,10 @@ from {{packageName}}.api_client import ApiClient
 from {{packageName}}.exceptions import (  # noqa: F401
     ApiTypeError,
     ApiValueError
-)
+){{#hasImport}}
+
```
Comment (MikeMcGarry) [inline python/generate/templates/api.mustache:17]: Is the purpose of this to create a blank line between these imports and those above? These imports are of the same group as the existing LUSID imports brought in above and according to PEP8 they should be grouped together with no blank line in between. https://www.python.org/dev/peps/pep-0008/#imports

I suggest removing the `{{#hasImport}}` and `{{/hasImport}}` tags to comply with PEP8 and simplify the template change.

File: python/generate/templates/api.mustache [hunk 2]
```diff
@@ -13,7 +13,10 @@ from {{packageName}}.api_client import ApiClient
...
+{{/hasImport}}{{#imports}}{{import}}
+{{/imports}}
```
Comment (MikeMcGarry) [inline python/generate/templates/api.mustache:19]: This seems to only bring in the imports required for this file which is amazing, nice work!

File: python/generate/templates/api.mustache [hunk 3]
```diff
@@ -107,10 +110,10 @@ class {{classname}}(object):
                               request; this effectively ignores the authentication
                               in the spec for a single request.
         :type _request_auth: dict, optional
-        :return: Returns the result object.
+        :return: Returns the result object, the HTTP status code, and the headers.
                  If the method is called asynchronously,
                  returns the request thread.
-        :rtype: {{#returnType}}tuple({{returnType}}, status_code(int), headers(HTTPHeaderDict)){{/returnType}}{{^returnType}}None{{/returnType}}
+        :rtype: {{#returnType}}({{returnType}}, int, HTTPHeaderDict){{/returnType}}{{^returnType}}None{{/returnType}}
```
Comment (MikeMcGarry) [inline python/generate/templates/api.mustache:116]: Looks great, not sure what the `status_code` and `headers` in the `rtype` where supposed to represent? This seems like a much more suitable approach and as you say is much more enjoyable to work with in PyCharm and likely other IDEs as well.

**Comment by MikeMcGarry** [comment]
Great work Marat, this is a fantastic addition to the type hints for the APIs. I think we can remove the `hasImports` tags to simplify the change and comply with PEP8, otherwise looks perfect.

**Review by MikeMcGarry — CHANGES_REQUESTED** [review]
Great work @mb-a, this is a fantastic addition to the type hints for the APIs. I think we can remove the hasImports tags to simplify the change and comply with PEP8, otherwise looks perfect.

File: python/generate/templates/api.mustache (hunk 1 above)
Comment (mb-a) [reply python/generate/templates/api.mustache:17]: Yes indeed, it was an erroneous stylistic choice!

File: python/generate/templates/api.mustache (hunk 2 above)
Comment (mb-a) [reply python/generate/templates/api.mustache:19]: I can't take all the credit, I was inspired by the C# generators :)

**Review by mb-a — COMMENTED** [review]
Removed non-PEP8 compliant spacing, excited for this to be deployed in prod!

**Pull request merged**